### PR TITLE
fix and improve token-program.md,token-program-advance.md  and close-mint.md

### PR DESCRIPTION
**Pull request by wuuer**
#### token-program.md
no try/catch for async/await calls, which is not following CONTRIBUTING.md.
[Token Extensions Program] link is wrong.
[add a second account on devnet] link is wrong.
wrong ts command in the section [Create an Associated Token Account to store the tokens].
console.log displays a wrong explorer link due to "!".
#### token-program-advance.md
[creating NFTs with Metaplex] link is wrong.
[previous chapter] link is wrong.
[Transfering Tokens] link is wrong.
The import @solana/web3.js statements in the section Brun Tokens are wrong.
The return value type "web3.Transaction" is wrong in the section Brun Tokens.
Missing delegate account for demonstration for the section Delegating tokens.
A wrong argument passes to calling the revoke function in the typescript for Revoke Delegate.
Missing some instructions for Delegating tokens, Revoke Delegate, and Burn Tokens.
#### close-mint.md
TERMINOLOGY "mint" is not proper in the context "In this lab, we'll create a mint with the `close mint` extension.".
missing typescript code example for using an account on devnet while calling "initializeKeypair" in the section [2. Run validator node]
missing typescript code example for importing our
new function in the section [3. Create a mint with close authority]
"mintInfo" is redeclared in the typescript.
#### token-program.md
add try/catch for async/await calls following CONTRIBUTING.md.
delete unnecessary "!" for generating a wrong explorer link.
add some missing instructions.
fix a wrong ts command for the section [Create an Associated Token Account to store the tokens].
#### token-program-advance.md
Delete "web3." for "web3.Transaction" in the section Brun Tokens.
Using the system program as the delegate in the typescript of Delegating tokens.
Add some instructions for Delegating tokens, Revoke Delegate, and Burn Tokens.
Fix [creating NFTs with Metaplex] link.
Fix [previous chapter] link.
Fix [Transferring Tokens] link.
Fix the import @solana/web3.js statements in the section Brun Tokens.
Fix a wrong argument passed to calling the revoke function in the typescript for Revoke Delegate.
#### close-mint.md
change "mint" to "token mint account".
add a typescript code example for using an account on devnet while calling "initializeKeypair" in the section [2. Run validator node]
add a typescript code example for importing our new function in the section [3. Create a mint with close authority]
change "const mintInfo" to "let mintInfo" because "mintInfo" is used again in the "CLOSE MINT" code part.

### CODE REVIEW

**Comment by mikemaccana** [comment]
Thanks @wuuer ! Just double checking - is this for the superteam lesson updates bounty?

File: content/courses/tokens-and-nfts/token-program-advanced.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ description:
 
 Finally, we'll cover burning tokens, and delegation. You may not use these in
 your own application, so if you're really excited about NFTs, feel free to skip
-to [creating NFTs with Metaplex](/content/courses/tokens/nfts-with-metaplex)!
+to
+[creating NFTs with Metaplex](/content/courses/tokens-and-nfts/nfts-with-metaplex)!
```
Comment (mikemaccana) [inline content/courses/tokens-and-nfts/token-program-advanced.md:16]:
```suggestion
[creating NFTs with Metaplex](/content/courses/tokens-and-nfts/nfts-with-metaplex.md)!
```

Thanks! We do links in a slightly odd way, see https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#links

Please go through and add the extensions on all the links - I know it's odd!

File: content/courses/tokens-and-nfts/token-program-advanced.md [hunk 2]
```diff
@@ -292,35 +375,58 @@ console.log(
...
+Replace `YOUR_TOKEN_MINT_ADDRESS_HERE` with your mint token address obtained
+from the previous chapter.
+
+Run the script using `npx esrun burn-tokens.ts`. You should see:
+
+```bash
+✅ Burn Transaction: https://explorer.solana.com/tx/P9JAK7bSAhycccGunDEThgt12QFiqMr9oexenEmRXXKoXsLKr2x64k9BWNppjTxFeVMUYjBEncRKe3gZsyd29JY?cluster=devnet
+```
+
+Open the explorer link, you will see the burn information.
```
Comment (mikemaccana) [inline content/courses/tokens-and-nfts/token-program-advanced.md:428]:
```suggestion
Open the Explorer link, you will see the burn information.
```

File: content/courses/tokens-and-nfts/token-program.md [hunk 3]
```diff
@@ -707,14 +725,17 @@ Open the token account in Solana Explorer. Look at the owner - it's the account
...
 #### Mint Tokens
 
 Now that we have a token mint and a token account, let's mint tokens to the
 token account. Recall that we set the `user` as the `mintAuthority` for the
 `mint` we created.
 
-Create a function `mintTokens` that uses the `spl-token` function `mintTo` to
-mint tokens:
+Create an empty file called `mint-tokens.ts`. Then uses the `spl-token` function
+`mintTo` to mint tokens. Remember to substitute in your token mint address and
```
Comment (mikemaccana) [inline content/courses/tokens-and-nfts/token-program.md:737]:
```suggestion
`mintTo()` to mint tokens. Remember to substitute in your token mint address and
```

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Thanks @wuuer this is a really comprehensive set of fixes. Just fix the links as mentioned, and update https://github.com/Unboxed-Software/solana-token-frontend - I suggest just making a replacement with `create-solana-dapp` per the bounty.

File: content/courses/tokens-and-nfts/token-program.md (hunk 3 above)
Comment (wuuer) [reply content/courses/tokens-and-nfts/token-program.md]: Could you please confirm whether you are suggesting that we add "()" to all function references?

Comment (mikemaccana) [reply content/courses/tokens-and-nfts/token-program.md]: It's a bit of convention in programming docs but no, it's not mandatory according to our current guidelines.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Hi @wuuer! Just a reminder to get the unboxed repos updated as discussed.

**Comment by wuuer** [comment]
> Hi @wuuer! Just a reminder to get the unboxed repos updated as discussed.

@mikemaccana Okay.

**Comment by mikemaccana** [comment]
@wuuer one more reminder! You have a good chance of winning this one if you get the other repos updated.

**Comment by wuuer** [comment]
> @wuuer one more reminder! You have a good chance of winning this one if you get the other repos updated.

# token and nfts
[token-program](https://github.com/solana-foundation/developer-content/pull/534)
# token extensions
[close-mint](https://github.com/solana-foundation/developer-content/pull/536)

**Comment by wuuer** [comment]
> > @wuuer one more reminder! You have a good chance of winning this one if you get the other repos updated.
>
@mikemaccana 
> # token and nfts
> [token-program](https://github.com/solana-foundation/developer-content/pull/534)
> 
> # token extensions
> [close-mint](https://github.com/solana-foundation/developer-content/pull/536)

**Pull request closed without merging**